### DISCUSSION TITLE
[feature-request] Allow list of variants for route params

### DISCUSSION THREAD

**Discussion by xaota** [remix-run/react-router#12175] [Proposals]
```ts
{
  path: "route/:param/:type?",
  Component,
  errorElement: <ErrorBoundary />
}
```

How i can describe variants for `:param` and `:type`?

I need show `errorElement` if param in my route not in variants array

Feature Request

```ts
{
  path: "route/:param/:type?",
  variants: {
    ":param": ["feature-a", "feature-b", "feature-c"],
    ":type": ["type-a", "type-b", "type-c"]
  },
  Component,
  errorElement: <ErrorBoundary />
}
```

And, if param not in `variants[paramName]` list -> return errorElement

**Comment by Bessonov** [comment]
+1. In an [unreleased version](https://github.com/Bessonov/node-http-router/tree/next?tab=readme-ov-file#endpointmatcher) of my (nodejs/server-side) router, [I am using some sort of validator for this](https://github.com/Bessonov/node-http-router/tree/next?tab=readme-ov-file#validators):

```ts
router.addRoute({
	matcher: endpoint('GET', /^\/user\/(?<userId>[^/]+)$/
		{
			url: {
				userId: chain(getNthVal(), requiredVal(), toNumVal()),
			},
			query: {
				// profile query parameter is optional in this example
				profile: chain(atLeastOneVal([undefined, 'short', 'full']), getNthVal()),
			},
		},
	),
	handler: ({ match: { result: { pathname, query } } }) => {
		return `User id: ${pathname.params.userId}, profile: ${query.profile}.`,
	}
})
```

They are more flexible than just `variants`.

**Reply by xaota** [reply]
This is imperative :)
